x/compliance/handler_test.go

```diff
@@ -1187,6 +1443,86 @@ func TestHandler_CertifyRevokedModelThatWasCertifiedEarlier(t *testing.T) {
 	}
 }
 
+func TestHandler_CDCertificateIDUpdateChangesOnlyOneComplianceInfo(t *testing.T) {
+	setup := Setup(t)
+
+	// add model version
+	vid1, pid1, softwareVersion1, softwareVersionString1 := setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion, testconstants.SoftwareVersionString)
+
+	// certify first model version
+	certifyModelMsg := NewMsgCertifyModel(
+		vid1, pid1, softwareVersion1, softwareVersionString1, dclcompltypes.ZigbeeCertificationType, setup.CertificationCenter)
+	_, err := setup.Handler(setup.Ctx, certifyModelMsg)
+	require.NoError(t, err)
+
+	// add second model version
+	vid2, pid2, softwareVersion2, softwareVersionString2 := setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion+1, testconstants.SoftwareVersionString)
+
+	// certify second model version
+	certifyModelMsg = NewMsgCertifyModel(
+		vid2, pid2, softwareVersion2, softwareVersionString2, dclcompltypes.ZigbeeCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, certifyModelMsg)
+	require.NoError(t, err)
+
+	// Update compliance info of first model version
+	updateComplianceInfoMsg := NewMsgUpdateComplianceInfo(setup.CertificationCenter, vid1, pid1, softwareVersion1, softwareVersionString1, dclcompltypes.ZigbeeCertificationType, 3, "", testconstants.ParentChild1)
+	updateComplianceInfoMsg.CDCertificateId += "new"
+	_, err = setup.Handler(setup.Ctx, updateComplianceInfoMsg)
+	require.NoError(t, err)
+
+	firstComplianceInfo, err := queryComplianceInfo(setup, vid1, pid1, softwareVersion1, testconstants.CertificationType)
+	require.NoError(t, err)
+
+	secondComplianceInfo, err := queryComplianceInfo(setup, vid2, pid2, softwareVersion2, testconstants.CertificationType)
+	require.NoError(t, err)
+
+	require.Equal(t, updateComplianceInfoMsg.CDCertificateId, firstComplianceInfo.CDCertificateId)
+	require.Equal(t, testconstants.CDCertificateID, secondComplianceInfo.CDCertificateId)
+}
+
```

Comment:
Please add the following test:
- there is Model1 for certificationID=X
- there is Model2 for certificationID=X
- we update Model2 to the same certicationID=X (set it explicitly) 
- check that query by certificationID=X returns the same value (the same array) as before